packages/vertica-nodejs/lib/connection.js

```diff
@@ -100,8 +101,15 @@ class Connection extends EventEmitter {
         case 'S': // Server supports TLS connections, continue with a secure connection
           break
         case 'N': // Server does not support TLS connections
-          self.stream.end()
-          return self.emit('error', new Error('The server does not support TLS connections'))
+          if (self.tls_mode == 'prefer') {
+            self.attachListeners(self.stream)
+            self.emit('sslconnect')
```

Comment:
For the management of our connection state events, 'sslconnect' is the proper event to emit. This may seem odd because we are not actually connecting with SSL wrapped socket in this case, however the sslconnect event just symbolizes that we completed our attempt at an ssl connection. 

So this code and this emission of 'sslconnect' is correct. Just wanted to add some context